build_openstudio_gems.rb

```diff
@@ -97,7 +100,38 @@ def make_package(install_dir, tar_exe, expected_ruby_version, bundler_version)
   system_call("#{bundle_cmd} config")
   puts "=" * 80
 
-  system_call("#{bundle_cmd} install")
+  puts "=" * 16 + " precaching packages and building them manually " + "=" * 16
+  # Shenigans onto shenanigans: Bundler, when you specify :git or :path
+  # actually installs in #{install_dir}/ruby/#{ruby_gem_dir} under the
+  # bundler/gems folder, and not under the gems/ one
+  # and that's causing a lot of troubles for openstudio to find them
+  # So here we:
+  #   * Download everything into the vendor/cache folder
+  #   * Find everything that's a directory, not a .gem file, go in there, build
+  #   it and move it back up
+  #   * Then we bundle install --local to use that vendor/cache,
+  #   and all is _well_
+  system_call("#{bundle_cmd} cache --no-install") # No need for "--all" conanfile sets BUNDLE_CACHE_ALL, and specifying it prints an annoying warning
```

Comment:
cache but not install/

This populates vendor/cache with .gem for stuff we get on rubygems, and folders for the :git and :path ones.